### ISSUE TITLE
Editor Freeze when working in Markdown

### ISSUE THREAD

**Issue by Stanzilla** [atom/atom#12104]
### Prerequisites

* [ ] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode)?
* [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
* [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
* [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
* [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
* [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

The only thing I could find was https://github.com/atom/atom/issues/11503 which was closed as fixed by the author.

For more information on how to write a good [bug report](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report) or [enhancement request](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion), see the `CONTRIBUTING` guide.

### Description

When editing a certain link markup, the editor freezes for a few seconds

### Steps to Reproduce

1. Make a MD file with `[[http://www.wowace.com/addons/weakauras/pages/display-types/#w-default-position-options|Default Position Options]]`
2. Select `Default Position Options]` and cut it
3. Paste that in the front of the string and remove the extra `[`
4. Remove the extra `]` at the end and the pipe

**Expected behavior:** [What you expected to happen]

no freeze

**Actual behavior:** [What actually happened]

editor freezes for a few seconds

### Versions
1.9.0.beta0
You can get this information from executing `atom --version` and `apm --version` at the command line. Also, please include the OS and what version of the OS you're running.

Video: https://dl.dropboxusercontent.com/s/26me77ovvqbwl2v/2016-07-05_18-53-52.webm?dl=0

**Comment by lee-dohm** [comment]
I tried this multiple times using Atom v1.9.0-beta0 on Mac OS X 10.11.5. I followed exactly what appears in the video using both mouse-only and keys-only navigation. I also included the extra formatting in case that had something to do with it (list item and bold around the link). None of the various attempts replicated the problem you're describing.

Did you try to replicate this in Safe Mode? To launch Safe Mode:

1. Completely exit all instances of Atom
1. Launch Atom with the command `atom --safe`
